Sources/OpenAPIRuntime/EventStreams/ServerSentEventsDecoding.swift

```diff
@@ -133,8 +157,16 @@ extension ServerSentEventsDeserializationSequence.Iterator {
         /// The current state of the state machine.
         private(set) var state: State
 
+        
+        /// A closure that determines whether the given byte sequence is the terminating byte sequence defined by the API.
+        /// - Parameter: A sequence of byte chunks.
+        /// - Returns: `True` until the terminating byte sequence is received.
+        let predicate: (ArraySlice<UInt8>) -> Bool
```

Comment:
The closure will cause the `State` enum to lose its conformance to `Hashable`. I don’t think there is an easy workaround for this, is there?

Reply:
Try to remove the conformance completely, it might only be used in tests. If it's used there, please add a manual implementation of Equatable/Hashable, that just ignores the closure in equality/hashing. I don't believe we ever use equality checking of the state in production code.

Reply:
Brilliant! simply removing it appears to have worked.